### PR TITLE
Extra checks for Honda Bosch Longitudinal

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: board/safety/safety_honda.h [hunk 1]
```diff
@@ -290,6 +299,13 @@ static int honda_tx_hook(CAN_FIFOMailBox_TypeDef *to_send) {
     }
   }
 
+  // Only tester present ("\x02\x3E\x80\x00\x00\x00\x00\x00") allowed on diagnostics address
+  if (addr == 0x18DAB0F1) {
+    tx &= (GET_BYTE((to_send), 0) == 0x02);
+    tx &= (GET_BYTE((to_send), 1) == 0x3E);
+    tx &= (GET_BYTE((to_send), 2) == 0x80);
```
Comment (adeebshihadeh) [inline board/safety/safety_honda.h:306]: check the rest of the bytes?
```suggestion
  if ((GET_BYTES_04(to_send) != 0x023E8000) || (GET_BYTES_48(to_send) != 0x0)) {
    tx = 0;
  }
```

Comment (pd0wm) [reply board/safety/safety_honda.h:306]: We already check the length is 2, so the exact padding shouldn’t matter.

Comment (adeebshihadeh) [reply board/safety/safety_honda.h:306]: hmm, the addr checks have the length as 8
https://github.com/commaai/panda/blob/12fccafc47109fc9e514f680dfe722a12ebe6823/board/safety/safety_honda.h#L13

Comment (pd0wm) [reply board/safety/safety_honda.h:306]: The message is iso-tp, so the first byte is the length (2 bytes). The other 5 bytes are just padding, so the contents are ignored by the receiver (some senders pad with `\x00`, some with `\xaa` I think),

Comment (pd0wm) [reply board/safety/safety_honda.h:306]: But if you're more comfortable constraining the whole message that's fine with me too!

Comment (adeebshihadeh) [reply board/safety/safety_honda.h:306]: yeah, think it's best to make the safety code as unambiguous as possible

**Pull request merged**